### PR TITLE
Bump Golang to v1.19.6 in v0.11.x line

### PR DESCRIPTION
**Pull request by sethiyash**
Bumping Golang to v1.19.6 in v0.11.x line

### CODE REVIEW

**Comment by rohitagg2020** [comment]
You are not changing it in `release.yml`. That file still points to go `1.18`. Maybe you want to cherrypick this [PR](https://github.com/carvel-dev/secretgen-controller/commit/c81677ee43b75ff716ad5cb64c2e6e8773592e9b#diff-87db21a973eed4fef5f32b267aa60fcee5cbdf03c67fafdc2a9b553bb0b15f34) if only diffs in your cherrypick you get is for golang version?

**Comment by praveenrewar** [comment]
+1 to what @rohitagg2020 mentioned. We should cherry pick [this](https://github.com/carvel-dev/secretgen-controller/pull/268/commits/bcecd3905d5654849caf7e939ae8094b77cb30f8) commit.

**Comment by sethiyash** [comment]
Cherry-picked the mentioned commit. Thanks, @rohitagg2020 @praveenrewar for pointing it out.

File: .github/workflows/golangci-lint.yml [hunk 1]
```diff
@@ -21,8 +21,13 @@ jobs:
       - name: Set up Go 1.19.5
         uses: actions/setup-go@v3
         with:
+<<<<<<< HEAD
```
Comment (praveenrewar) [inline .github/workflows/golangci-lint.yml]: Seems like some conflicts need to be resolved.

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM golang:1.19.5 AS deps
+FROM --platform=$BUILDPLATFORM golang:1.19.6 AS deps
```
Comment (praveenrewar) [inline Dockerfile:1]: Seems like this change is coming up from the supporting arm64 build. I don't think we want this in this PR.

**Review by neil-hickey — APPROVED** [review]
LGTM

**Pull request closed without merging**